opa-manager-operator/tests/integration/test_charm.py

```diff
@@ -0,0 +1,115 @@
+import logging
+from pathlib import Path
+import os
+import time
+import pytest
+import yaml
+import re
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+from jinja2 import Template
+
+log = logging.getLogger(__name__)
+
+meta = yaml.safe_load(Path("metadata.yaml").read_text())
+
+KUBECONFIG = os.environ.get("TESTING_KUBECONFIG", "~/.kube/config")
+
+config.load_kube_config(KUBECONFIG)  # TODO: check how to use this with designated file
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test):
+    charm = await ops_test.build_charm(".")
+    # Due to: https://github.com/juju/python-libjuju/issues/515
+    #  We have to use the k8s API to wait, we cannot use:
+    # ops_test.model.applications['gatekeeper'].units[0].workload_status
+    model_name = ops_test._default_model_name
+    role_binding_file = Path("/tmp/k8s-rolebinding.yaml")
+    with open("docs/gatekeeper-rb.yaml.template", "r") as fh:
+        template = Template(fh.read())
+        role_binding_file.write_text(
+            template.render(
+                service_account_user=f"system:serviceaccount:{model_name}:kubernetes-operator"
```

Comment:
The user name here needs to match the application name:

```suggestion
                service_account_user=f"system:serviceaccount:{model_name}:opa-manager-test-operator"
```